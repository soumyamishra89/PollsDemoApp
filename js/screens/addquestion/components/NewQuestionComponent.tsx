import React, { useState } from 'react';
import { ScrollView, View, StyleSheet } from 'react-native';
import { TextInput, Title, Button } from 'react-native-paper';
import { useSafeArea } from 'react-native-safe-area-context';
import Styles from '../../../styles';
import Appbar from '../../../screens/common/compoonent/Appbar';
import colors from '../../../styles/colors';
import { NavigationProps } from '../../../data/types/ScreenProps';

interface Props extends NavigationProps {
    addNewQuestion: (question: string, choices: string[]) => Promise<boolean>
}

export default function NewQuestionComponent(props: Props) {
    const [newQuestion, setNewQuestion] = useState('');
    const [choice1, setChoice1] = useState('');
    const [choice2, setChoice2] = useState('');
    const [choice3, setChoice3] = useState('');
    const [choice4, setChoice4] = useState('');

    const [loading, setLoading] = useState(false);

    const addNewQuestion = async () => {
        setLoading(true);
        const postSuccess = await props.addNewQuestion(newQuestion, [choice1, choice2, choice3, choice4]);
        if (postSuccess) {
            props.navigation.goBack();
        } else {
            setLoading(false);
        }
    }

    const disableButton = !newQuestion || !choice1 || !choice2 || !choice3 || !choice4;
    const inset = useSafeArea();
    
    return (
        <View style={[Styles.matchParent, {paddingBottom: inset.bottom}]}>
            <Appbar title="New Qeestion" />
            <ScrollView style={Styles.contentContainer} contentContainerStyle={{flex: 1}}>
                <TextInput
                    label='New question'
                    mode="outlined"
                    value={newQuestion}
                    onChangeText={setNewQuestion}
                    multiline
                    theme={textInputTheme}
                />
                {
                    // CHoices section
                }
                <Title style={styles.title}>Enter your choices</Title>
                <TextInput
                    label='Choice 1'
                    value={choice1}
                    onChangeText={setChoice1}
                    style={styles.choiceText}
                    theme={textInputTheme}
                />
                <TextInput
                    label='Choice 2'
                    value={choice2}
                    onChangeText={setChoice2}
                    style={styles.choiceText}
                    theme={textInputTheme}
                />
                <TextInput
                    label='Choice 3'
                    value={choice3}
                    onChangeText={setChoice3}
                    style={styles.choiceText}
                    theme={textInputTheme}
                />
                <TextInput
                    label='Choice 4'
                    value={choice4}
                    onChangeText={setChoice4}
                    style={styles.choiceText}
                    underlineColor={colors.primary}
                    theme={textInputTheme}
                />
            </ScrollView>
            <Button mode="contained" onPress={addNewQuestion} disabled={disableButton} style={[Styles.button, disableButton && Styles.disabledButton]} loading={loading}>Vote</Button>
        </View>
    )
}

const textInputTheme: any = {colors: {primary: colors.primary}}
const styles = StyleSheet.create({
    title: {
        color: colors.primary,
        marginTop: 16
    },
    choiceText: {
        marginVertical: 8
    }
})