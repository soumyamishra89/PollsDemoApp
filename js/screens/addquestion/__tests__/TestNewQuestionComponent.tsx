import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NewQuestionComponent from '../components/NewQuestionComponent';

describe("NewQuestionComponent", () => {
    describe("Snapshot test", () => {
        it("With question data", () => {
            const newQuestionTree = renderer.create(<NewQuestionComponent navigation={{}} addNewQuestion={jest.fn()} />).toJSON();
            expect(newQuestionTree).toMatchSnapshot();

        })
    })
})