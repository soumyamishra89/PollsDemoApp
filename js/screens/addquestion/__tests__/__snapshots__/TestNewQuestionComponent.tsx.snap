// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestionComponent Snapshot test With question data 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "height": "100%",
        "width": "100%",
      },
      Object {
        "paddingBottom": 20,
      },
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#3C7AFF",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3C7AFF",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            Array [
              false,
              Object {
                "alignItems": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          accessible={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "fontSize": 17,
                },
                undefined,
              ],
            ]
          }
        >
          New Qeestion
        </Text>
      </View>
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
    </View>
  </RCTSafeAreaView>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    style={
      Object {
        "flex": 1,
        "marginTop": 16,
        "paddingHorizontal": 16,
      }
    }
  >
    <View>
      <View
        style={Object {}}
      >
        <View>
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#f6f6f6",
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "minHeight": 64,
                "paddingBottom": 0,
                "paddingTop": 8,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "bottom": 0,
                    "height": 2,
                    "left": 10,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 8,
                  }
                }
              />
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                New question
              </Text>
              <Text
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#3C7AFF",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                New question
              </Text>
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 1,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                New question
              </Text>
            </View>
            <TextInput
              allowFontScaling={true}
              editable={true}
              multiline={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              selectionColor="#3C7AFF"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "textAlign": "left",
                    "zIndex": 1,
                  },
                  Object {},
                  Object {
                    "paddingBottom": 22,
                    "paddingTop": 22,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlignVertical": "top",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                Object {
                  "color": "#3C7AFF",
                  "marginTop": 16,
                },
              ],
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                Object {
                  "color": "#3C7AFF",
                  "marginTop": 16,
                },
              ],
            ],
          ]
        }
      >
        Enter your choices
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "marginVertical": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#3C7AFF",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Choice 1
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Choice 1
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#3C7AFF"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "paddingHorizontal": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 4,
                  "paddingTop": 24,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "marginVertical": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#3C7AFF",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Choice 2
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Choice 2
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#3C7AFF"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "paddingHorizontal": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 4,
                  "paddingTop": 24,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "marginVertical": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#3C7AFF",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Choice 3
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Choice 3
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#3C7AFF"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "paddingHorizontal": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 4,
                  "paddingTop": 24,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "marginVertical": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#3C7AFF",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#3C7AFF",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Choice 4
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Choice 4
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#3C7AFF"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "paddingHorizontal": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 4,
                  "paddingTop": 24,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#D8D8D8",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "marginBottom": 16,
        "marginHorizontal": 16,
        "minWidth": 64,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.32)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Vote
        </Text>
      </View>
    </View>
  </View>
</View>
`;
